internal/resources/slo/resource_slo.go

```diff
@@ -552,6 +568,20 @@ func packAlerting(tfAlerting map[string]interface{}) slo.SloV00Alerting {
 		SlowBurn:    &tfSlowBurn,
 	}
 
+	// All options in advanced options will be optional
+	// Adding a second feature will need to make a better way of checking what is there
+	if failures := tfAlerting["advanced_options"]; failures != nil {
+		lf := failures.([]interface{})
+		if len(lf) > 0 {
+			lf2 := lf[0].(map[string]interface{})
```

Comment:
issue: this will panic if the runtime value of `lf[0]` fails the type assertion.

Don't you need to use the `lf2, ok := lf[0].(map[string]interface{})` version and check `ok`?

Reply:
Interesting point. Yeah I should fix that.

Reply:
We actually do this exact same thing for the alerting struct as a whole, the terraform can panic if it doesn't match the expected format. Think I should switch that too?

Reply:
yes, if it's easy. If we can handle more gracefully than panic, we should.

where else are we doing it though? The only uncheecked type assertions I see in this function are on line 574 and 576

Reply:
I actually misunderstood. the alerting struct handles it all gracefully. I've replicated it now  here.